sw/device/tests/aon_timer_wdog_bite_reset_test.c

```diff
@@ -0,0 +1,92 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+/**
+ * Execute the aon timer wdog bite reset test.
+ */
+static void execute_test(dif_aon_timer_t *aon_timer, dif_pwrmgr_t *pwrmgr,
+                         uint32_t bark_time_us) {
```

Comment:
The difs which uses these pointers doesn't define them as const, if I do so I would need to cast them. So I think I'll leave it this way.

Reply:
I think the only DIF functions that don't have `const` are `dif_<peripheral>_init`, if I am not mistaken, the rest will never change the contents of `dif_<peripheral>_t`, and hence should always be `const` pointers.

The only places that I can see don't use `const` are functions in the testutils that you use in this code.

It could be a nice clean-up to add `const` to them (fairly minimal) change.

This is really a minor thing, but just to be consistent it could be a nice little tidy-up.

Reply:
You're right, is `aon_timer_testutils_wakeup_config` functions that is not defining it as const.

Reply:
It's up to you, if you decide to go for it, I think it makes sense in a separate clean-up PR.

Reply:
Done in a different commit.